apps/system/js/wrapper_factory.js

```diff
@@ -96,37 +96,48 @@
       // Use fake origin for named windows in order to be able to reuse them,
       // otherwise always open a new window for '_blank'.
       var origin = null;
-      if (name == '_blank') {
-
-        // If we already have a browser and we receive an open request,
-        // display it in the current browser frame.
-        var activeApp = Service.query('AppWindowManager.getActiveWindow');
-        if (activeApp && (activeApp.isBrowser() || activeApp.isSearch())) {
-          activeApp.navigate(url);
-          return;
-        }
+      switch (name) {
+        case 'pinned':
```

Comment:
Sounds good to me. Actually, didn't add the '_' just because of that. It didn't sound too generic for using it as a keyword... '_samescope' sounds good. Thanks!